crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -142,24 +387,621 @@ impl<F, T> TryFrom<ResponseRouterData<F, JpmorganPaymentsResponse, T, PaymentsRe
     }
 }
 
-//TODO: Fill the struct with respective fields
-// REFUND :
-// Type definition for RefundRequest
-#[derive(Default, Debug, Serialize)]
-pub struct JpmorganRefundRequest {
-    pub amount: StringMinorUnit,
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganCaptureRequest {
+    capture_method: Option<CapMethod>,
+    merchant: Option<MerchantCapReq>,
+    recurring: Option<RecurringCapReq>,
+    installment: Option<InstallmentCapReq>,
+    payment_method_type: Option<PaymentMethodTypeCapReq>,
+    ship_to: Option<ShipToCapReq>,
+    initiator_type: Option<String>,
+    account_on_file: Option<String>,
+    original_transaction_id: Option<String>,
+    is_amount_final: Option<bool>,
+    amount: MinorUnit,
+    currency: Option<common_enums::Currency>,
+    merchant_order_number: Option<String>,
+    risk: Option<RiskCapReq>,
+    retail_addenda: Option<RetailAddendaCapReq>,
+    account_holder: Option<AccountHolderCapReq>,
+    statement_descriptor: Option<String>,
+    partial_authorization_support: Option<String>,
+    payment_request_id: Option<String>,
+    multi_capture: Option<MultiCapture>,
+    sub_merchant_supplemental_data: Option<SubMerchantSupplementalData>,
 }
 
-impl<F> TryFrom<&JpmorganRouterData<&RefundsRouterData<F>>> for JpmorganRefundRequest {
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SubMerchantSupplementalData {
+    pub custom_data: Option<CustomData>,
+    pub service_address: Option<ServiceAddress>,
+    pub business_information: Option<BusinessInformation>,
+    pub partner_service: Option<PartnerService>,
+    pub shipping_info: Option<ShippingInfo>,
+    pub recurring_billing: Option<RecurringBilling>,
+    pub merchant_reported_revenue: Option<MerchantReportedRevenue>,
+    pub order_information: Option<OrderInformation>,
+    pub consumer_device: Option<ConsumerDevice>,
+    pub merchant_identification: Option<MerchantIdentification>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CustomData {
+    pub external_transaction_reference_number: Option<String>,
+    pub external_transaction_type: Option<String>,
+    pub external_merchant_id: Option<String>,
+    pub merchant_order_reference_id: Option<String>,
+    pub external_batch_id: Option<String>,
+    pub merchant_expected_deposit_date: Option<String>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ServiceAddress {
+    pub line1: Option<String>,
+    pub line2: Option<String>,
+    pub city: Option<String>,
+    pub state: Option<String>,
+    pub postal_code: Option<String>,
+    pub country_code: Option<String>,
+}
```

Comment:
Wrap sensitive fields within `Secret` please. Refer to other connectors' code to understand what can be included in plaintext and what needs to be masked.

Reply:
alright